src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -311,6 +324,26 @@ private static Deployment updateDeployment (Request req, Response res) {
             List<String> versionIds = versionsToInsert.stream().map(v -> v.id).collect(Collectors.toList());
             Persistence.deployments.updateField(deploymentToUpdate.id, "feedVersionIds", versionIds);
         }
+
+        // If updatedDocument has deleted a CSV file, also delete that CSV file from S3
+        if (updateDocument.containsKey("peliasCsvFiles")) {
+            List<String> csvUrls = (List<String>) updateDocument.get("peliasCsvFiles");
+            // Only delete if the array differs
+            if (!csvUrls.equals(deploymentToUpdate.peliasCsvFiles)) {
+                for (String existingCsvUrl : deploymentToUpdate.peliasCsvFiles) {
```

Comment:
As mentioned in the companion UI PR, I get a NPE on this line when uploading a file.

Reply:
Extra check added in 1f5673d390acd5a8af7ad68b32260b14e5afe526. Please let me know if this fixes things!